gateway/src/main/java/com/mx/path/gateway/connect/filter/TracingFilter.java

```diff
@@ -20,27 +20,30 @@
 public class TracingFilter extends RequestFilterBase {
   @Override
   public final void execute(Request request, Response response) {
-    Span span = CustomTracer.startSpan(request.getTraceKey());
-    if (span == null) {
+    Tracer tracer = GlobalTracer.get();
+
+    if (tracer == null) {
```

Comment:
What case would cause tracer to be null?

Reply:
There is a very small chance that none of the correct headers get sent in and the datadog agent that is running next to the service does not catch the request on the way in and we could not have a tracer.

Reply:
Thank you for the explanation!